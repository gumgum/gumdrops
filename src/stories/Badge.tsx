import React, { ReactChild, ReactChildren } from 'react';
import cx from 'classnames';
import { Colors } from 'types';

export interface BadgeProps extends React.HTMLAttributes<Element> {
    text?: JSX.Element[] | ReactChildren | ReactChild;
    color?: Colors;
    empty?: boolean;
}

const baseClass = 'gds-badge';

export const Badge: React.FC<BadgeProps> = ({
    color = 'default',
    text,
    empty,
    className,
    ...otherProps
}) => {
    const rootClass = cx(baseClass, className, {
        [`${baseClass}--${color}`]: color,
        [`${baseClass}--empty`]: empty
    });

    return (
        <span className={rootClass} {...otherProps}>
            {!empty && text}
        </span>
    );
};
