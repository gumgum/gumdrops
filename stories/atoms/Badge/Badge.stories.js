import React from 'react';

import Badge from '../../../components/atoms/Badge';

const options = [
    '', 'inverse', 'success', 'success-inverse', 'info', 'info-inverse', 'warning', 'warning-inverse', 'danger', 'danger-inverse'
];

export default {
  title: 'Atoms/Badge',
  component: Badge,
  parameters :{
    docs: {
        description: {
            component: 'The `<Badge>` component is a numerical indicator of associated items. For a simple colored circle without a number inside, pass in the "empty" prop. If empty, it will not display any text within the badge.'
        }
    }
  },
  argTypes: {
    text: { control: 'text'},
    empty: {control: 'boolean'},
    context: {
      options,
      control: { type: 'select' }
    },
  },
};

const Template = (args) => 
    <Badge { ...args} />
;
  
  export const Default = Template.bind({});
  Default.args = {
    context: '',
    text: '2'
  };

// const options = {
//     'No Value': '',
//     inverse: 'inverse',
//     success: 'success',
//     'success-inverse': 'success-inverse',
//     info: 'info',
//     'info-inverse': 'info-inverse',
//     warning: 'warning',
//     'warning-inverse': 'warning-inverse',
//     danger: 'danger',
//     'danger-inverse': 'danger-inverse'
// };

// export default {
//     component: Badge,
//     title: 'Atoms/Badge',
//     decorators: [withKnobs],
//     parameters: {
//         docs: { page: mdx }
//     }
// };

