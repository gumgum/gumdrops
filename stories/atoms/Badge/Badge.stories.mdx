import { Story, Props } from '@storybook/addon-docs';
import Badge from '../../../components/atoms/Badge';

# Badge

The `<Badge>` component is a numerical indicator of associated items. For a simple colored circle without a number inside, pass in the "empty" prop. If empty, it will not display any text within the badge. 

## Component Example
<Story
    id="atoms-badge--default"
/>

## Props
<Props of={Badge} />
