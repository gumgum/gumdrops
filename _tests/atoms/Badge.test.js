import renderer from 'react-test-renderer';
import React from 'react';
import Badge from '../../components/atoms/Badge';

test('Expect <Badge> to render properly', () => {
    const props = {
        text: 'This is text',
        context: 'inverse',
        className: 'foo-class',
        style: {
            width: '1000px'
        }
    };
    const tree = renderer
        .create(
            <Badge {...props}>
                <div>foo</div>
            </Badge>
        )
        .toJSON();
    expect(tree).toMatchSnapshot();
});

test('Expect <Badge> to render properly with empty', () => {
    const props = {
        text: 'Ignored text',
        context: 'inverse',
        className: 'foo-class',
        empty: true,
        style: {
            width: '1000px'
        }
    };
    const tree = renderer.create(<Badge {...props} />).toJSON();
    expect(tree).toMatchSnapshot();
});
