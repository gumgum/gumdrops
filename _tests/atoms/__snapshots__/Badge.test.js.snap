// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Expect <Badge> to render properly 1`] = `
<span
  className="gds-badge foo-class gds-badge--inverse"
  style={
    {
      "width": "1000px",
    }
  }
>
  This is text
</span>
`;

exports[`Expect <Badge> to render properly with empty 1`] = `
<span
  className="gds-badge foo-class gds-badge--inverse gds-badge--empty"
  style={
    {
      "width": "1000px",
    }
  }
/>
`;
